addons/toolbars/src/utils/create-cycle-value-array.ts

```diff
@@ -0,0 +1,7 @@
+import type { ToolbarItem } from '../types';
+
+export const createCycleValueArray = (items: ToolbarItem[]) => {
+  // Do not allow items in the cycle arrays that are conditional in placement
+  const valueArray = items.filter((item) => !item.condition).map((item) => item.value);
```

Comment:
what is condition? what are the use cases?

storybook is divided into a preview iframe and a manager UI. the toolbars are declared in the preview but displayed in the manager. serializing functions across the channel is possible if the functions are simple/stateless, but we prefer not to do it for a variety of reasons. if the condition use cases are not common i would consider removing this additional complexity

Reply:
Use case is in the preview.js for official-storybook with the Locale global

You will now be able to add the complex "reset" functionality that for instance addon-a11y has to make it's own drop down to handle (the vision simulator) - such an item can not be in the cycle-able list as it's behaviour is not static.

I can remove it for sure - but I baked it in as I see this functionality being achieved outside of the toolbar addon, but still trying to achieve the toolbar addon functionality (so let's get people back into the herd was my idea)

Will remove if you still think so after this comment :-)

Reply:
I'd vote to remove this. If we must keep it, I'd propose a non-function like `isReset: true`. Or maybe even automatically infer it from the `value: ''` which is a "reset" value.

Reply:
Then I bind myself to being able to foresee all future applications and usage of this feature

By just opening up for a function (which should also send api along come to think of it) that returns a boolean I remove myself from the equation

If this is not in then the whole ability to hide/show an item has to be removed to cater for reliable cycling via keys as you would be able to cycle through an item that is currently not there.

Reply:
fixed to the much better type approach